salt/cloud/clouds/proxmox.py

```diff
@@ -585,6 +586,23 @@ def create(vm_):
     if not wait_for_state(vmid, 'running'):
         return {'Error': 'Unable to start {0}, command timed out'.format(name)}
 
+    # For QEMU VMs, we can get the IP Address from qemu-agent
+    if nodeType == 'qemu':
+        # We have to wait for a bit for qemu-agent to start
+        log.debug("Waiting for qemu-agent to start...")
+        time.sleep(20)
```

Comment:
I don't like this, we should just use the `cloud.wait_for_fun` utils module to know when the node is available and ready to return.

https://github.com/saltstack/salt/blob/2018.3/salt/cloud/clouds/openstack.py#L713  Here is an example.